### PR TITLE
Homepage animations and times

### LINKED ISSUE #195: New refresh animation on homescreen

### PR DESCRIPTION
**Pull request by patrickfeltes**
Closes #195 #192 
- Adds new pull-down refresh animation
- Adds animated countdown tickers
- Adds time of the first event to the upcoming events section

### CODE REVIEW

File: app/src/main/java/org/hackillinois/android/database/entity/Event.kt [hunk 1]
```diff
@@ -37,10 +37,10 @@ data class Event(
         val minutes = eventEndTime.get(Calendar.MINUTE)
 
         return when {
-            hour == 0 -> String.format("12:%02d AM", minutes)
-            hour < 12 -> String.format("%d:%02d AM", hour, minutes)
-            hour == 12 -> String.format("12:%02d PM", minutes)
-            else -> String.format("%d:%02d PM", hour % 12, minutes)
+            hour == 0 -> String.format("12:%02dam", minutes)
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/database/entity/Event.kt]: dam.

Comment (bstrauch24) [reply app/src/main/java/org/hackillinois/android/database/entity/Event.kt]: this is some nice code

File: app/src/main/java/org/hackillinois/android/view/custom/CustomRefreshView.kt [hunk 2]
```diff
@@ -1,88 +1,39 @@
...
 import org.hackillinois.android.R
 
 class CustomRefreshView constructor(
     context: Context?,
     attrs: AttributeSet? = null,
     defStyleAttr: Int = 0
-) : View(context, attrs, defStyleAttr), IRefreshStatus {
-
-    private val ANIMATION_DURATION = 888
-    private val starfish: Drawable = resources.getDrawable(R.drawable.starfish)
```
Comment (bstrauch24) [inline app/src/main/java/org/hackillinois/android/view/custom/CustomRefreshView.kt]: :'(

**Comment by patrickfeltes** [comment]
> Looks good! Did you write all the ticker code from scratch by the way? I thought you were using a library that had everything

Yeah it's mostly from scratch. Some of the animation code came from here: https://github.com/Vinayrraj/Android-FlipDigitView. Existing libraries only seemed to work for single digits and some didn't allow font size and color changes.

**Pull request merged**